client.go

```diff
@@ -345,10 +345,10 @@ func filterCloseErr(err error) error {
 	switch {
 	case err == nil:
 		return nil
-	case err == io.EOF:
-		return ErrClosed
 	case errors.Cause(err) == io.EOF:
 		return ErrClosed
+	case errors.Cause(err) == io.ErrUnexpectedEOF:
```

Comment:
Should we switch to using `errors.Is()` / `errors.As()` ?

Reply:
agree